conformance/tests/httproute-request-percentage-mirror.go

```diff
@@ -0,0 +1,236 @@
+	Features: []features.FeatureName{
+		features.SupportGateway,
+		features.SupportHTTPRoute,
+		// features.SupportHTTPRouteRequestPercentageMirror,
+	},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		var (
+			ns      = "gateway-conformance-infra"
+			routeNN = types.NamespacedName{Name: "request-percentage-mirror", Namespace: ns}
+			gwNN    = types.NamespacedName{Name: "same-namespace", Namespace: ns}
+			gwAddr  = kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+		)
+
+		kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
+
+		testCases := []http.ExpectedResponse{
+			{
+				Request:   http.Request{Path: "/percent-mirror"},
+				Namespace: ns,
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Path: "/percent-mirror",
+					},
+				},
+				Backend: "infra-backend-v1",
+				MirroredTo: []http.MirroredBackend{
+					{
+						BackendRef: http.BackendRef{
+							Name:      "infra-backend-v2",
+							Namespace: ns,
+						},
+						Percent: ptrTo(int32(20)),
+					},
+				},
+			}, {
+				Request:   http.Request{Path: "/percent-mirror-fraction"},
+				Namespace: ns,
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Path: "/percent-mirror-fraction",
+					},
+				},
+				Backend: "infra-backend-v1",
+				MirroredTo: []http.MirroredBackend{
+					{
+						BackendRef: http.BackendRef{
+							Name:      "infra-backend-v2",
+							Namespace: ns,
+						},
+						Percent: ptrTo(int32(50)),
+					},
+				},
+			}, {
+				Request: http.Request{
+					Path: "/percent-mirror-and-modify-headers",
+					Headers: map[string]string{
+						"X-Header-Remove":     "remove-val",
+						"X-Header-Add-Append": "append-val-1",
+					},
+				},
+				ExpectedRequest: &http.ExpectedRequest{
+					Request: http.Request{
+						Path: "/percent-mirror-and-modify-headers",
+						Headers: map[string]string{
+							"X-Header-Add":        "header-val-1",
+							"X-Header-Add-Append": "append-val-1,header-val-2",
+							"X-Header-Set":        "set-overwrites-values",
+						},
+					},
+					AbsentHeaders: []string{"X-Header-Remove"},
+				},
+				Namespace: ns,
+				Backend:   "infra-backend-v1",
+				MirroredTo: []http.MirroredBackend{
+					{
+						BackendRef: http.BackendRef{
+							Name:      "infra-backend-v2",
+							Namespace: ns,
+						},
+						Percent: ptrTo(int32(35)),
+					},
+				},
+			},
+		}
+
+		for i := range testCases {
+			expected := testCases[i]
+			t.Run(expected.GetTestCaseName(i), func(t *testing.T) {
+				// Assert request succeeds before doing our distribution check
+				http.MakeRequestAndExpectEventuallyConsistentResponse(t, suite.RoundTripper, suite.TimeoutConfig, gwAddr, expected)
+
+				// Override to not have more requests than expected
+				suite.TimeoutConfig.RequiredConsecutiveSuccesses = 1
```

Comment:
This change will affect other tests as well. Why do we need to do so?

Reply:
oh man.. there is a whole story behind this change. 
Essentially since I test mirroring, I started to see more requests that I had actually sent. so the mirror expectations were totally wrong. I learned to find this config the hard way.. 

Any other suggestion here if this is not desired?

Reply:
yes, this var defaults to 3: https://github.com/kubernetes-sigs/gateway-api/blob/082cc90333d45ab4e95d5b28cb9e3e3e27f2ecbb/conformance/utils/config/timeout.go#L128

Patching this in place is problematic, as it changes the user configuration and affects many other tests as well. Can we instead copy the whole `TimeoutConfig` to a new var and change that to our desired value? It should be feasible, as `http.MakeRequestAndExpectEventuallyConsistentResponse` accepts a Config as a parameter.

Reply:
Awesome, thanks for that. added this.